.github/workflows/build_push_service.yml

```diff
@@ -6,20 +6,169 @@ name: sc-app-build-push
+          echo "Version of antivirus_service is '${AVCHECK_VERSION}' (as Imagetag only for Master/Release)"  
+          echo "GITHUB_SHA is '${GITHUB_SHA}' (Imagetag only for Master/Release)"      
+      - name: cancel workflow if Dockertag is empty String
+        if: ${{ env.DOCKER_TAG_POSTFIX == '' }}
+        uses: andymckay/cancel-action@0.2
+      - name: Echo DOCKER_TAG_POSTFIX
+        run: echo "DOCKER_TAG_POSTFIX is ${DOCKER_TAG_POSTFIX}"
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to DockerHub
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+      - name: Build and push Scanfile
+        uses: docker/build-push-action@v2.2.1
+        with:
+          context: .
+          file: ./Dockerfile.scanfile
+          push: true
+          tags: ${{ env.DOCKER_NAMESPACE }}/${{ env.AVSCANFILE_REPO }}:${{ env.DOCKER_TAG_POSTFIX }}
+      - name: Build and push Scanurl
+        uses: docker/build-push-action@v2.2.1
+        with:
+          context: .
+          file: ./Dockerfile.scanurl
+          push: true
+          tags: ${{ env.DOCKER_NAMESPACE }}/${{ env.AVSCANURL_REPO }}:${{env. DOCKER_TAG_POSTFIX}}
+      - name: Build and push Webserver
+        uses: docker/build-push-action@v2.2.1
+        with:
+          context: .
+          file: ./Dockerfile.webserver
+          push: true
+          tags: ${{ env.DOCKER_NAMESPACE }}/${{ env.AVSCANWEBSERVER_REPO }}:${{ env.DOCKER_TAG_POSTFIX }}
+      - name: Build and push SHA Tag for Master and Release on Scanfile
+        if: ${{ github.event.inputs.branch-prefix == 'master' || github.event.inputs.branch-prefix == 'release' }}
+        uses: docker/build-push-action@v2.2.1
+        with:
+          context: .
+          file: ./Dockerfile.scanfile
+          push: true
+          tags: ${{ env.DOCKER_NAMESPACE }}/${{ env.AVSCANWEBSERVER_REPO }}:${{ env.DOCKER_TAG_SHA }}
+      - name: Build and push SHA Tag for Master and Release on Scanurl
+        if: ${{ github.event.inputs.branch-prefix == 'master' || github.event.inputs.branch-prefix == 'release' }}
+        uses: docker/build-push-action@v2.2.1
+        with:
+          context: .
+          file: ./Dockerfile.scanurl
+          push: true
+          tags: ${{ env.DOCKER_NAMESPACE }}/${{ env.AVSCANWEBSERVER_REPO }}:${{ env.DOCKER_TAG_SHA }}
+      - name: Build and push SHA Tag for Master and Release on Webserver
+        if: ${{ github.event.inputs.branch-prefix == 'master' || github.event.inputs.branch-prefix == 'release' }}
+        uses: docker/build-push-action@v2.2.1
+        with:
+          context: .
+          file: ./Dockerfile.webserver
+          push: true
+          tags: ${{ env.DOCKER_NAMESPACE }}/${{ env.AVSCANWEBSERVER_REPO }}:${{ env.DOCKER_TAG_SHA }}
```

Comment:
To mutch steps with to offen build, why where the SHA tags not push with this bevore?

Reply:
1. conditions in actions are limited. So 5 braches with 3 images.
2. we had SHA tags before an dev's and devops demanded to keep SHA Tags and enhanced epic / naming convention.